x-pack/plugins/cloud_security_posture/public/pages/findings/utils.ts

```diff
@@ -103,3 +105,35 @@ export const useBaseEsQuery = ({
 
   return baseEsQuery;
 };
+
+export const addFilter = ({
+  filters,
+  dataView,
+  field,
+  value,
+  negate,
+}: {
+  filters: Filter[];
+  dataView: FindingsBaseProps['dataView'];
+  field: string;
+  value: Serializable;
+  negate: boolean;
+}): Filter[] => {
+  const dataViewField = dataView.getFieldByName(field);
+  if (!dataViewField) return filters;
+
+  const singleValue = Array.isArray(value) ? value[0] : value;
+
+  const filter = buildFilter(
+    dataView,
+    dataViewField,
+    FILTERS.PHRASE,
+    negate,
+    false,
+    singleValue,
+    null,
+    FilterStateStore.APP_STATE
+  );
+
+  return [...filters, filter].filter(isNonNullable);
+};
```

Comment:
a few things to note about this function: 
- it returns a `Filter[]` which is what we need to plug into our `filters` URL query
- we build that filter using a given field and value
- for simplicity, i've reduced array values (like rule.section to a its 1st member, we can/should change this) 
- the field name we pass must be the actual key on the document. this, along with performance requirements (replace `_source` with `fields`) are the reasons fields in this PR changed to their actual names (`resource_subtype` ->`resource.sub_type`).